### PR TITLE
PowerVS: Update listCOSInstances to continue querying

### PR DESCRIPTION
**Pull request by hamzy**
ListResourceInstancesWithContext does not return every matching result in one call.  Therefore, continue querying until the NextURL is empty.

### CODE REVIEW

File: pkg/destroy/powervs/cloudobjectstorage.go [hunk 1]
```diff
@@ -147,7 +213,7 @@ func (o *ClusterUninstaller) destroyCOSInstance(item cloudResource) error {
 	response, err = o.controllerSvc.DeleteResourceInstanceWithContext(ctx, options)
 
 	if err != nil && response != nil && response.StatusCode != http.StatusNotFound {
-		return errors.Wrapf(err, "failed to delete COS instance %s", item.name)
+		return fmt.Errorf("failed to delete COS instance %s %w", item.name, err)
```
[the remark this answers is not in the document]
Comment (hamzy) [reply pkg/destroy/powervs/cloudobjectstorage.go:216]: Done

File: pkg/destroy/powervs/cloudobjectstorage.go [hunk 2]
```diff
@@ -29,41 +28,108 @@ func (o *ClusterUninstaller) listCOSInstances() (cloudResources, error) {
...
+
+			o.Logger.Debugf("listCOSInstances: NO matching COS instance against: %s", o.InfraID)
+			for _, instance := range resources.Resources {
+				o.Logger.Debugf("listCOSInstances: only found COS instance: %s", *instance.Name)
+			}
+
+			if resources.NextURL != nil {
+				start, err := resources.GetNextStart()
+				if err != nil {
+					o.Logger.Debugf("listCOSInstances: err = %w", err)
```
[the remark this answers is not in the document]
Comment (hamzy) [reply pkg/destroy/powervs/cloudobjectstorage.go:122]: Good catch!
```
./test8.go:32:2: (*github.com/sirupsen/logrus.Logger).Debugf does not support error-wrapping directive %w
```

**Review by r4f4 — COMMENTED** [review]
/approve
I cannot comment on neither test the functionality itself but the code looks sane to me.

**Pull request merged**